pkg/distributors/contracts/MerkleOrchard.sol

```diff
@@ -63,16 +63,57 @@ contract MerkleOrchard {
     ) internal {
         uint256[] memory amounts = new uint256[](tokens.length);
 
+        // To save gas when setting claimed statuses in storange we group updates
+        // into currentBits for a particular channel, only setting them when a claim
+        // on a new channel is seen, or on the final iteration
+
+        // for aggregating claims
+        uint256 currentBits;
+        bytes32 currentChannelId;
+        uint256 currentWordIndex;
+        uint256 currentClaimAmount;
+
         Claim memory claim;
-        IERC20 token;
         for (uint256 i = 0; i < claims.length; i++) {
             claim = claims[i];
-            token = tokens[claim.tokenIndex];
 
-            require(!isClaimed(token, claim.distributor, claim.distribution, liquidityProvider), "cannot claim twice");
+            // When we process a new claim we either
+            // a) aggregate the new claim bit with previous claims of the same channel/claim bitmap
+            // b) set claim status and start aggregating a new set of currentBits for a new channel/word
+            if (currentChannelId == bytes32(0)) {
+                currentChannelId = keccak256(abi.encodePacked(address(tokens[claim.tokenIndex]), claim.distributor));
```

Comment:
I think we need something like

```
function _getChannelId(IERC20 token, address distributor) private returns (bytes32) {
 return keccak256(abi.encodePacked(address(token, distributor))
}
```
As this is popping up a bunch of times and causing line breaks

Reply:
yeah, I was encountering stack too deep orders, adding that function would probably be a better way to avoid them